matchzoo/data_generator/pair_data_generator.py

```diff
@@ -0,0 +1,34 @@
+"""Base generator."""
+
+import math
+import typing
+import numpy as np
+
+from matchzoo import reorganize_data_pack_pair_wise
+from .data_generator import DataGenerator
+
+
+class PairDataGenerator(DataGenerator):
+    """Generate pair-wise data...."""
+    def __init__(self, data_pack, num_dup: int=1, num_neg: int=1, **kwargs):
+        """:class:`PairDataGenerator` constructor...."""
+        self._num_dup = num_dup
+        self._num_neg = num_neg
```

Comment:
`num_neg` is only used to calculate `steps` so just initialize a `self._steps` here.